### ISSUE TITLE
Content-Type header missing with codable routing

### ISSUE THREAD

**Issue by djones6** [IBM-Swift/Kitura#1188]
When invoking a route that uses codable routing, JSON data is returned in the message body, but the `Content-Type` header is missing.

Example project:  https://github.com/djones6/CodableTest

Here's the route and the response:
```swift
    router.get("/codable") {
        (completion: ([Hello]?, RequestError?) -> Void) in
        let hello = Hello(message: "Hello World")
        completion([hello], nil)
    }
```
```
$ curl -i http://localhost:8080/codable
HTTP/1.1 200 OK
Date: Thu, 23 Nov 2017 15:53:40 GMT
Content-Length: 27
Connection: Keep-Alive
Keep-Alive: timeout=60

[{"message":"Hello World"}]
```
The equivalent using the existing API does include the header:
```swift
    router.get("/json") {
        request, response, _ in
        let hello = Hello(message: "Hello World")
        try response.send(json: [hello]).end()
    }
```
```
$ curl -i http://localhost:8080/json
HTTP/1.1 200 OK
Date: Thu, 23 Nov 2017 15:53:48 GMT
Content-Type: application/json
Content-Length: 27
Connection: Keep-Alive
Keep-Alive: timeout=60

[{"message":"Hello World"}]
```

**Comment by seabaylea** [comment]
This has just been reported by @svanimpe as well.

In each of the "safely" functions we do the encoding directly, e.g.  for `postSafely()`:
https://github.com/IBM-Swift/Kitura/blob/master/Sources/Kitura/CodableRouter.swift#L235
```swift
    let encoded = try JSONEncoder().encode(result)
    response.status(.created)
    response.send(data: encoded)
```
we have two choices:
1) Add a call to `response.headers.setType("json")`
2) Call `response.send(result)` instead and delegate the encoding to the send call

Approach #2 makes the most sense as there's no duplication of code/function. Currently there's no way to tell the response which encoder to use - that's not a problem at the moment, but we'll need to provide a setter in the future once we support other encoding types.

**Comment by tunniclm** [comment]
@ddunn2 I think this issue is solved by https://github.com/IBM-Swift/Kitura/pull/1190

**Comment by ddunn2** [comment]
@tunniclm I'll have a read through of the item you've linked, thanks! :)

**Comment by rolivieri** [comment]
@ddunn2 @tunniclm Yes, @sandmman noticed this while implementing the two new proposals and has added code to address this issue.

**Comment by christiancompton** [comment]
Reassigning to @sandmman, to be closed once this aspect of #1190 is complete.

**Comment by ianpartridge** [comment]
Fixed in #1190
